### DISCUSSION TITLE
Autocomplete : can't get command name

### DISCUSSION THREAD

**Discussion by xNiux** [discord-net/Discord.Net#2820] [Q&A] [answered]
Hi, 

I would like to add autocompletion features on slashed command.
But When I catch InteractionCreated event from the client, I doesn't have any property in  SocketInteraction.Data.

Here interaction task :  

![image](https://github.com/discord-net/Discord.Net/assets/25155398/c224afd9-2003-4430-8c25-7275b1546adb)

But I have this error : 

![image](https://github.com/discord-net/Discord.Net/assets/25155398/b856774e-c17a-46fa-baab-6bf593a7ea98)

When I run the application (Without the line with error), the Data property is ok : 

![image](https://github.com/discord-net/Discord.Net/assets/25155398/a73454c2-c171-44b6-95a4-c24191c5cd08)

What is the problem ?

**Comment by exsersewo** [comment]
The first issue is, you're trying to get the object `cmdName` when you define the data as `cmdData`. 

Secondly, that's not how you'd get the Command Name from an autocomplete. 

What you want to do is cast the `Data` property to [`SocketAutocompleteInteractionData`](https://discordnet.dev/api/Discord.WebSocket.SocketAutocompleteInteractionData.html) then you can get it from `autoCompleteData.CommandName`.

As `Discord.API.AutocompleteInteractionData` is an internal class.

**Comment by xNiux** [comment]
First ... You're write for the object name : Bad simplification for the screenshot ^^

And so, I tried to cast into SocketAutocompleteInteractionData, but, if the code is OK for the parser ... At execution, I have an exception : 

With hard cast : 

Code :
![image](https://github.com/discord-net/Discord.Net/assets/25155398/5c69d135-9afc-40e9-8ed1-9071f2dab9c7)

Result :
![image](https://github.com/discord-net/Discord.Net/assets/25155398/96039954-8583-467e-a562-7ffa543d9f89)



With "as" function : 

![image](https://github.com/discord-net/Discord.Net/assets/25155398/5e233d3c-f64a-40c8-b47c-c11585aa81cb)

**Reply by exsersewo** [reply, marked as answer]
Ah my bad, try casting `interaction` to a `SocketAutocompleteInteraction` instead, `Data` will be auto converted

**Reply by xNiux** [reply]
You're right ! thanks a lot !
